### DISCUSSION TITLE
Resetting individual fields in useFieldArray

### DISCUSSION THREAD

**Discussion by efreila** [react-hook-form/react-hook-form#4352] [Q&A]
I'm currently trying to use useFieldArray. I got it working by following numerous examples, but I'm running into an issue when I try to have dependent select inputs in a specific item in the returned "fields" array. 

When I loop over the fields returned from the hook, I return a custom component:
```
{fields.map((item, index) => {
    return (
        <li key={item.id}>
            <FilterInputGroup
              index={index}
              data={data!.getUniqueValuesForFilter}
              item={item}
              control={control}
              remove={remove}
              watch={watch}
            />
       </li>
   );
})}`
```

The FilterInputGroup component is a collection of three react-select components.

When a user modifies the selected option in the first select input, it should automatically clear the selected value in the third select input (if there is one).

I tried combining using  the value returned from watch (`const selectedField: FilterFieldValues = watch(`filters[${index}].field`);` with useEffect to detect for changes, but each time I tried to use setValue or reset in the effect, it wouldn't work properly. setValue only worked when I set it to a value in my options array (didnt reset to empty), and reset keeps giving me an infinite loop. 

Is there a better approach to take for achieving this functionality? I essentially want all of my FilterInputGroup components to listen for changes to a specific input and reset their own specific fields when a change occurs (don't want to touch other FilterInputGroup components).

**Comment by bluebill1049** [comment]
can you show a simple codesandbox?

**Reply by efreila** [reply]
Sure, I tried my best to recreate what I'm trying to go for and left some comments where I'm having the issue.

https://codesandbox.io/s/beautiful-einstein-l7fym?file=/src/App.js

Edit: I debugged further and it seems like I got it working using setValue() and setting the other two fields to null, but my UI doesn't update since setValue() doesn't trigger a rerender. I figured out a hacky solution by passing in my watch() value as the key prop to the components that I'm trying to rerender, but I feel like there is probably a better way to do this.

Edit 2: using setValue() on a name that contains an index is also a problem because if you delete an item before it in the fields array, it triggers the useEffect (due to index change), and resets the fields in all the other field items.

**Reply by otrebu** [reply]
Hey @efreila I admit I didn't analyse your example in great details, but I had a similar issue in resetting a value within an array field.

After some experimentation I found this method:

`control.updateFormState();`

Which does trigger a re-render for me.
